Today Extension/TodayViewController.swift

```diff
@@ -0,0 +1,226 @@
+import Alamofire
+import SwiftyJSON
+
+ @objc(TodayViewController) class TodayViewController: UIViewController, NCWidgetProviding {
+
+    var routesTable: UITableView = UITableView()
+    var favorites: [String] = []
+    var coordinates: [String] = []
+    var routes: [Route?] = []
+
+    var didFetchRoutes: Bool = false
+
+    let group = DispatchGroup()
+    var locationManager: CLLocationManager!
+    var currentLocation: CLLocationCoordinate2D?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        extensionContext?.widgetLargestAvailableDisplayMode = .compact
+
+        setUpLocation()
+        setUpRoutesTableView()
+        view.addSubview(routesTable)
+        createConstraints()
+
+        favorites = TodayExtensionManager.shared.retrieveFavoritesNames()
+        coordinates = TodayExtensionManager.shared.retrieveFavoritesCoordinates()
+
+        searchForRoutes()
+    }
+
+    func searchForRoutes() {
+        if
+            favorites.count > 0,
+            let start = currentLocation {
+                Network.getMultiRoutes(startCoord: start, time: Date(), endCoords: coordinates, endPlaceNames: favorites) { (request) in
+                    self.processRequest(request: request)
+                }
+            }
+    }
+
+    func processRequest(request: APIRequest<MultiRoutesRequest, Error>) {
+        request.performCollectingTimeline { (response) in
+            switch response.result {
+            case .success(let routesResponse):
+                self.routes = routesResponse.data
+                self.rearrangeRoutes()
+                self.didFetchRoutes = true
+                self.extensionContext?.widgetLargestAvailableDisplayMode = .expanded
+                self.routesTable.reloadData()
+            case .failure(let networkError):
+                if let error = networkError as? APIError<Error> {
+                    self.processRequestError(error: error)
+                }
+            }
+        }
+    }
+
+    func processRequestError(error: APIError<Error>) {
+        let title = "Network Failure: \((error.error as NSError?)?.domain ?? "No Domain")"
+        let description = (error.localizedDescription) + ", " + ((error.error as NSError?)?.description ?? "n/a")
+
+        routes = []
+        print("Error Title: \(title)")
+        print("Error Description: \(description)")
+    }
+
+    /// Called in response to the user tapping the “Show More” or “Show Less” buttons
+    func widgetActiveDisplayModeDidChange(_ activeDisplayMode: NCWidgetDisplayMode, withMaximumSize maxSize: CGSize) {
+        let expanded = activeDisplayMode == .expanded
+        preferredContentSize = expanded ? CGSize(width: maxSize.width, height: 110.0 * CGFloat(favorites.count)) : maxSize
+
+        routesTable.reloadData()
+    }
+
+    func createConstraints() {
+        routesTable.snp.makeConstraints {(make) in
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(110.0*5)
```

Comment:
Definitely make these constants, especially like `numberOfCells`. Wherever this should be used, this should be made a variable